pytorch_lightning/metrics/bleu.py

```diff
@@ -0,0 +1,143 @@
+from collections import Counter
+import math
+import re
+
+
+def math_log(num: int):
+    return -99999999 if num == 0.0 else math.log(num)
+
+
+def best_match_len(translate_list: list, reference_list: list):
+    """Finding the closet length of the reference texts with respect to translated text
+
+    Args:
+        translate_list: A list of translated text
+        reference_list: A list of reference texts
+
+    Return:
+        ref_len: the best match length of text of the reference lists with translated text
+        translate_len: the length of translated text
+    """
+    translate_len = len(translate_list)
+    len_diff_lists = [abs(len(ref.split()) - translate_len) for ref in reference_list]
+    best_match_len_index = len_diff_lists.index(min(len_diff_lists))
+    ref_len = len(reference_list[best_match_len_index].split())
+
+    return ref_len, translate_len
+
+
+def count_ngram(ngram_input_list: list, num_grams: int) -> Counter:
+    """Counting how many times each word appears in a given text with ngram
+
+    Args:
+        ngram_input_list: A list of translated text or reference texts
+        num_grams: gram value ranged 1 to 4
+
+    Return:
+        ngram_counter: Counter object of ngram
+    """
+    ngram_counter = Counter()
+
+    for i in range(1, num_grams + 1):
```

Comment:
is there a more efficient way to do this than two nested for loops?

Reply:
Sorry, I can't think of other than nested for loops. Can you help for a more efficient way?